Root object types with only one `UncheckedArray[T]` field generate invalid C code

### What happened?

```nim
type
  Obj = object
    f: UncheckedArray[byte]

let o = new Obj
```

gcc (11.3) reports (confusing error message)
```shell
error: flexible array member in a struct with no named members
```

clang (14.0) reports
```shell
error: flexible array member 'f' not allowed in otherwise empty struct
```

### Nim Version

0.20..devel

### Current Standard Output Logs

```shell
$ nim r test2.nim 
Hint: used config file '/home/user/Projects/Nim/config/nim.cfg' [Conf]
Hint: used config file '/home/user/Projects/Nim/config/config.nims' [Conf]
Hint: used config file '/home/user/.config/nim/config.nims' [Conf]
...................................................................
test2.nim(5, 5) Hint: 'o' is declared but not used [XDeclaredButNotUsed]
CC: Projects/Nim/lib/system/exceptions.nim
CC: Projects/Nim/lib/std/private/digitsutils.nim
CC: Projects/Nim/lib/system/dollars.nim
CC: Projects/Nim/lib/system.nim
CC: test2.nim
/home/user/.cache/nim/test2_d/@mtest2.nim.c:34:5: error: flexible array member in a struct with no named members
   34 | NU8 f[SEQ_DECL_SIZE];
      |     ^
Error: execution of an external compiler program 'gcc -c  -w -fmax-errors=3 -pthread   -I/home/user/Projects/Nim/lib -I/home/user -o /home/user/.cache/nim/test2_d/@mtest2.nim.c.o /home/user/.cache/nim/test2_d/@mtest2.nim.c' failed with exit code: 1


user@dd-laptop:~$ /home/user/.cache/nim/test2_d/@mProjects@sNim@slib@ssystem.nim.c:269:5: error: flexible array member in a struct with no named members
  269 | NU8 f[SEQ_DECL_SIZE];
      |     ^
```


### Expected Standard Output Logs

```shell
Compiles with error report by nim compiler.
```


### Possible Solution

forbid root object types with only one field which type is `UncheckedArray[T]`

### Additional Information

_No response_

--- a/compiler/ccgtypes.nim
+++ b/compiler/ccgtypes.nim
@@ -774,8 +774,15 @@ proc getRecordDesc(m: BModule; typ: PType, name: Rope,
     result = structOrUnion & " " & name
     result.add(getRecordDescAux(m, typ, name, baseType, check, hasField))
     let desc = getRecordFields(m, typ, check)
-    if desc == "" and not hasField:
-      result.addf("char dummy;$n", [])
+    if not hasField:
+      if desc == "":
+        result.add("\tchar dummy;\n")
+      elif typ.len == 1 and typ.n[0].kind == nkSym:
+        let field = typ.n[0].sym
+        let fieldType = field.typ.skipTypes(abstractInst)
+        if fieldType.kind == tyUncheckedArray:
+          result.add("\tchar dummy;\n")
+      result.add(desc)
     else:
       result.add(desc)
     result.add("};\L")

--- a/tests/ccgbugs/t20787.nim
+++ b/tests/ccgbugs/t20787.nim
@@ -1,0 +1,4 @@
+type
+  Obj = object
+    f: UncheckedArray[byte]
+let o = new Obj

